service/worker/migration/force_replication_workflow.go

```diff
@@ -141,6 +186,62 @@ func ForceReplicationWorkflow(ctx workflow.Context, params ForceReplicationParam
 	return workflow.NewContinueAsNewError(ctx, ForceReplicationWorkflow, params)
 }
 
+func maybeKickoffTaskQueueUserDataReplication(ctx workflow.Context, params ForceReplicationParams, onDone func(failureReason string)) error {
+	workflow.Go(ctx, func(ctx workflow.Context) {
+		taskQueueUserDataReplicationDoneCh := workflow.GetSignalChannel(ctx, taskQueueUserDataReplicationDoneSignalType)
+		var errStr string
+		// We don't care if there's more data to receive
+		_ = taskQueueUserDataReplicationDoneCh.Receive(ctx, &errStr)
+		onDone(errStr)
+	})
+
+	if workflow.GetVersion(ctx, taskQueueUserDataReplicationVersionMarker, workflow.DefaultVersion, 1) == workflow.DefaultVersion || params.ContinuedAsNewCount > 0 {
```

Comment:
can you explain the logic of the `params.ContinuedAsNewCount > 0`?

Reply:
I'll add a comment but basically we only want to start the child workflow once on startup.

Reply:
I see. is there any versioning concern with the goroutine above? I guess there's no problem with just listening on a signal?

Reply:
There's no problem but I might just move all of this logic into the versioned branch so the intent is clearer.